js/apps/account-ui/src/applications/Applications.tsx

```diff
@@ -128,15 +128,22 @@ const Applications = () => {
                 className="pf-u-align-items-center"
                 dataListCells={[
                   <DataListCell width={2} key={`client${application.clientId}`}>
-                    <Button
-                      className="pf-u-pl-0 title-case"
-                      component="a"
-                      variant="link"
-                      onClick={() => window.open(application.effectiveUrl)}
-                    >
-                      {application.clientName || application.clientId}{" "}
-                      <ExternalLinkAltIcon />
-                    </Button>
+                    {application.effectiveUrl && (
+                      <Button
+                        className="pf-u-pl-0 title-case"
+                        component="a"
+                        variant="link"
+                        onClick={() => window.open(application.effectiveUrl)}
```

Comment:
Although his code was was already there it can be slightly improved by passing a `href` and `target` prop instead. 
```suggestion
                        href={application.effectiveUrl}
                        target="_blank"
                        rel="noopener"
```